### PR TITLE
Complete Add ID to Patient search screen

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/164980438

### CODE REVIEW

File: app/src/main/res/layout/screen_addidtopatientsearch.xml [hunk 1]
```diff
@@ -21,8 +21,9 @@
       android:layout_marginTop="@dimen/spacing_16"
       android:ellipsize="end"
       android:maxLines="2"
-      android:text="@string/addidtopatientsearch_title"
-      android:textAppearance="@style/Clinic.V2.TextAppearance.Body0Left.White72" />
+      android:textAppearance="@style/Clinic.V2.TextAppearance.Body0Left.White100"
+
```
Comment (sanchita) [inline app/src/main/res/layout/screen_addidtopatientsearch.xml]: Extra line.

File: app/src/main/res/values/styles_v2.xml [hunk 2]
```diff
@@ -658,4 +654,18 @@
     <item name="android:paddingStart">@dimen/spacing_16</item>
     <item name="android:paddingTop">@dimen/spacing_16</item>
   </style>
+
+  <style name="Clinic.V2.TextAppearance.Identifier" />
```
Comment (sanchita) [inline app/src/main/res/values/styles_v2.xml]: I can see that this is a bold style.
Can we stick to the name used in Zeplin designs - > `Clinic.V2.TextAppearance.Numeric-Bold`

Comment (sanchita) [reply app/src/main/res/values/styles_v2.xml]: We don't need another `parent` style for `Clinic.V2.TextAppearance.Numeric-Bold`.
its parent is already `Clinic.V2.TextAppearance`.
So you can merge this with `Clinic.V2.TextAppearance.Identifier.Numeric` and rename it.

File: app/src/main/res/layout/screen_addidtopatientsearch.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    android:scrollbars="vertical">
+
+    <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:orientation="vertical"
+      android:paddingBottom="64dp"
+      android:paddingEnd="@dimen/spacing_20"
+      android:paddingStart="@dimen/spacing_24"
+      android:paddingTop="@dimen/spacing_16"
```
Comment (sanchita) [inline app/src/main/res/layout/screen_addidtopatientsearch.xml]: The `paddingTop` is `24dp` as per the Zeplin designs.

File: app/src/main/res/layout/screen_addidtopatientsearch.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    android:layout_below="@id/addidtopatientsearch_toolbar"
+    android:scrollbarStyle="insideOverlay"
+    android:scrollbars="vertical">
+
+    <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:orientation="vertical"
+      android:paddingBottom="64dp"
+      android:paddingEnd="@dimen/spacing_20"
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/res/layout/screen_addidtopatientsearch.xml]: Done.

File: app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt [hunk 5]
```diff
@@ -501,4 +505,138 @@ class PatientsScreenControllerTest {
...
+            bpPassportCode = "3a0053f2-2a08-48a3-9f1c-1f4f83ad99d1",
+            isPatientPresent = false,
+            shouldOpenAddIdToPatientSearchScreen = true
+        ),
+        testCase(
+            bpPassportCode = "d0941c47-a70f-4969-b36e-ee323ff6faaf",
+            isPatientPresent = false,
+            shouldOpenAddIdToPatientSearchScreen = true
+        )
+    ).subList(1, 2)
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt]: Whoops, forgot to remove this.

File: app/src/main/res/layout/screen_addidtopatientsearch.xml (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/res/layout/screen_addidtopatientsearch.xml]: Fixed.

File: app/src/main/res/values/styles_v2.xml (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/res/values/styles_v2.xml]: I have pulled in the new styles from Zeplin and amended the commit.

File: app/src/main/res/layout/screen_addidtopatientsearch.xml (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/res/layout/screen_addidtopatientsearch.xml]: Fixed.

File: app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt [hunk 6]
```diff
@@ -248,4 +253,28 @@ class PatientsScreenController @Inject constructor(
           }
         }
   }
+
+  private fun handleScannedBpPassportCodes(events: Observable<UiEvent>): Observable<UiChange> {
+    val screenCreated = events.ofType<ScreenCreated>()
+    val scannedBpPassportCodeStrema = events.ofType<PatientsScreenBpPassportCodeScanned>()
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt]: Done.

File: app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt [hunk 7]
```diff
@@ -248,4 +253,28 @@ class PatientsScreenController @Inject constructor(
...
+
+    val openPatientSummary = foundPatientStream
+        .filterAndUnwrapJust()
+        .map { patient -> { ui: Ui -> ui.openPatientSummary(patient.uuid) } }
+
+    val openAddIdToPatientSearchScreen = Observables
+        .combineLatest(foundPatientStream, scannedBpPassportCodeStrema)
+        .filter { (foundPatient, _) -> foundPatient is None }
+        .map { (_, scannedBpPassportCode) -> scannedBpPassportCode.bpPassportCode }
+        .map { scannedId -> Identifier(value = scannedId.toString(), type = BpPassport) }
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt]: Done.

**Comment by vinaysshenoy** [comment]
@sanchita You were planning to push something to this branch, right? Will you resolve the conflicts as well? It's a pretty straightforward fix.

**Pull request merged**